### DISCUSSION TITLE
S3Hook.delete_objects() not deleting files

### DISCUSSION THREAD

**Discussion by rsotogar** [apache/airflow#22747] [Q&A]
Hey all,

This method suddenly started erroring out with message: `errors when deleting: [path/to/data]`. The method will both error out inside a custom python callable and in the S3DeleteObjectsOperator. I had this working fine for over 5 months. Anyone knows what could happen?

Best,

**Comment by vicsteiner** [comment]
I see it's been a long time since the question was posted. I wonder if you have solved it?

I have observed an increase of a similar error recently in some of the DAGs I run daily.

Looking at the response json I see `'Message': 'We encountered an internal error. Please try again.'` but the method `S3Hook.delete_objects()` (https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/s3.py#L1304) has no parameter for retries. For now the solution I can see is to override (I am using it inside a custom operator mostly) the method and implement it with the retry option.
